### PR TITLE
Text attribute accesses optimization

### PR DESCRIPTION
**Pull request by R4PaSs**
As a small optimization, accesses to private attributes in Text are replaced by direct accesses as in `file` or `collection`.

On the compiler this yields between 3 and 4% improvement in performances, however in some programs like nitmd, this yields around 15% optimization.

### CODE REVIEW

**Comment by privat** [comment]
You should check the result of the sed, also check that you do not access two time the same attribute: use local variables to cache results

**Comment by R4PaSs** [comment]
Reroll, most (hopefully all) of the double underscores should be gone now, I also tried to reduce sequential accesses to a resource to a minimum where applicable

File: lib/core/text/flat.nit [hunk 1]
```diff
@@ -489,12 +496,12 @@ private class FlatStringByteView
...
 	do
-		# Check that the index (+ first_byte) is not larger than last_byte
+		# Check that the index (+ _first_byte) is not larger than _last_byte
 		# In other terms, if the index is valid
 		assert index >= 0
 		var target = self.target
-		assert (index + target.first_byte) <= target.last_byte
-		return target.items[index + target.first_byte]
+		assert (index + target._first_byte) <= target._last_byte
+		return target._items[index + target._first_byte]
```
Comment (privat) [inline lib/core/text/flat.nit]: factorize `index + target._first_byte`

**Comment by privat** [comment]
+1. Maybe, you should document in shortcut attributes that they are shortcut

**Comment by xymus** [comment]
+1

**Pull request merged**